### DISCUSSION TITLE
Resync interval for NTP clock

### DISCUSSION THREAD

**Discussion by rvdbreemen** [bxparks/AceTimeClock#4] [General]
So most NTP clock implementations use an forced ntp resyncafter a certain interval. So a default could be 60 minutes.

Then in the mainloop you could have a `ntpClock.loop();`, which then implements two things:

1. if timeNotSync then force resync, this way in case the initial setup fails you can retry. I would build in a backoff, like once every 60 seconds or so, you do not want to spam ntp servers
2. then check to see if it is time to resync NPT, if so, then force NTP sync.

This way, your ntp clock can be synced automatically.

**Comment by bxparks** [comment]
I think you want to look at `SystemClockLoop` and `SystemClockCoroutine` for retry logic.

**Comment by rvdbreemen** [comment]
Solved. Thanks for your pointers.

**Comment by bxparks** [comment]
I have just finished the refactoring of AceTime into AceTime/AceTimeClock. I'm going to transfer this to the AceTimeClock project.

**Comment by bxparks** [comment]
Moving this to Discussions since it's not a bug with the library.
